cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/MasterEnvironment.java

```diff
@@ -89,8 +91,10 @@ MasterEnvironmentRunnable createRunnable(MasterEnvironmentRunnableContext contex
 
   /**
    * Returns a {@link SparkConfig} of this environment.
+   *
+   * @param localizeResources localize resources needed for spark on master environment
    */
-  default SparkConfig getSparkSubmitConfig() {
+  default SparkConfig getSparkSubmitConfig(Map<String, SparkLocalizeResource> localizeResources) {
```

Comment:
It would be good to create a new class (ex: SparkSubmitContext) that gets passed in, and have a method in that class to get the resources to localize.

This would make the SPI more extensible, in case we need to pass any other information in the future we can just add a method to the class without changing this method signature.

Reply:
yea makes sense.